### PR TITLE
[SD-112] Resolve the issue of the site form layout crashing.

### PR DESCRIPTION
**Pull request by vincent-gao**
### Jira
https://digital-vic.atlassian.net/browse/SD-112

### Changes
This PR fixes the recurring layout breakdown issues caused by inconsistencies between active and inactive configurations, which are distributed across different modules.

- Introduced use of `Drupal\Component\Utility\NestedArray` for more robust config handling.
- Implemented a more thorough config update process that,
    - Reads the existing config
    - Merges it with new config changes
    - Deduplicates and sorts dependencies

The root cause of our layout issues came from configurations being spread across multiple modules (e.g. tide_site and tide_site_theming). When update hooks only modify configs in one module (e.g. tide_site), they can inadvertently overwrite settings from other modules (e.g. tide_site_theming), leading to inconsistencies and layout failures.

For instance, content-reference configurations exist in both tide_site and tide_site_theming. Updating only from tide_site can overwrite tide_site_theming settings, causing layout breakdowns.

To mitigate these issues, we should consolidate all field configurations for a module within its top-level module. This approach, combined with proper update methods, can significantly reduce layout disappearance problems. (tide mono-repo 😹 )

### CODE REVIEW

File: tide_site.install [hunk 1]
```diff
@@ -22,7 +24,7 @@ function tide_site_install() {
  * New field taxonomy image logo.
  */
 function tide_site_update_10001() {
-  module_load_include('inc', 'tide_core', 'includes/helpers');
```
Comment (vincent-gao) [inline tide_site.install:25]: we shouldn't use deprecated functions.

File: tide_site.install [hunk 2]
```diff
@@ -42,11 +44,20 @@ function tide_site_update_10001() {
     'core.entity_view_display.taxonomy_term.sites.default',
   ];
   foreach ($form_configs as $form_config) {
-    $config = \Drupal::configFactory()->getEditable($form_config);
-    $config_read = _tide_read_config($form_config, $config_location, FALSE);
-    $config->set('dependencies', $config_read['dependencies']);
-    $config->set('content', $config_read['content']);
-    $config->set('hidden', $config_read['hidden']);
-    $config->save();
```
Comment (vincent-gao) [inline tide_site.install:50]: Arbitrarily overwriting active configurations may lead to layout inconsistencies.

File: tide_site.install [hunk 3]
```diff
@@ -42,11 +44,20 @@ function tide_site_update_10001() {
...
+    $rewrite = _tide_read_config($form_config, $config_location, FALSE);
+    $display_config_entity = \Drupal::configFactory()->getEditable($form_config);
+    $original_config = $display_config_entity->getRawData();
+    $rewritten_config = NestedArray::mergeDeep($original_config, $rewrite);
+    if ($rewritten_config['dependencies']) {
```
Comment (vincent-gao) [inline tide_site.install:51]: The reason I need to handle the dependencies data seems to be due to a bug with `NestedArray::mergeDeep` (I'm not entirely sure), which causes data duplication within dependencies. I might need to check Drupal.org for more information

**Review by yeniatencio — APPROVED** [review]
LGTM.

**Pull request merged**